local/local.go

```diff
@@ -87,13 +102,98 @@ func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...Image
 		downloadBaseOnce: &sync.Once{},
 	}
 
-	for _, v := range ops {
-		image, err = v(image)
-		if err != nil {
-			return nil, err
-		}
+	image, err = processPreviousImageOption(image, imageOpts.prevImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = processBaseImageOption(image, imageOpts.baseImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = prepareNewWindowsImage(image)
```

Comment:
It feels a bit weird to read through this function and see that `prepareNewWindowsImage` is always called. Maybe we want to extract the guard conditionals out of the `prepare*` functions. E.G
```
if image.inspect.Os == "windows" {
    image, err = prepareNewWindowsImage(image)
    if err != nil {
        return nil, err
    }
}
```

Reply:
Yeah, I was feeling the same thing. The prior private `prepare*` functions have internal conditional guards too. Is it ok to have Windows be the weird one or should I look for a pattern to move the guards out of all of them?

Reply:
Yeah, good call. I'll take a pass at it. The `prepare*` function guards should be easy to move out.

Reply:
I moved all the guards up into `NewImage` for local and remote.